Git-commit: 1adebbd34f0fd43b9afc4de81565c1205aa32b44
From: Catalin Marinas <catalin.marinas@arm.com>
Date: Tue, 16 Mar 2021 10:27:53 +0800
Subject: [PATCH] of: unittest: Fix build on architectures without
 CONFIG_OF_ADDRESS
Patch-mainline: v5.11-rc1
References: openEuler-21.03

mainline inclusion
from mainline-5.11-rc1
commit aed5041ef9a3f594ed9dc0bb5ee7e1bbccfd3366
category: bugfix
bugzilla: 50423
CVE: NA

---------------------------------------------

of_dma_get_max_cpu_address() is not defined if !CONFIG_OF_ADDRESS, so
return early in of_unittest_dma_get_max_cpu_address().

Fixes: 07d13a1d6120 ("of: unittest: Add test for of_dma_get_max_cpu_address()")
Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Catalin Marinas <catalin.marinas@arm.com>
Signed-off-by: Jing Xiangfeng <jingxiangfeng@huawei.com>
Reviewed-by: Liu Shixin <liushixin2@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 drivers/of/unittest.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/of/unittest.c b/drivers/of/unittest.c
index 98cc0163301b..eb51bc147440 100644
--- a/drivers/of/unittest.c
+++ b/drivers/of/unittest.c
@@ -874,6 +874,9 @@ static void __init of_unittest_dma_get_max_cpu_address(void)
 	struct device_node *np;
 	phys_addr_t cpu_addr;
 
+	if (!IS_ENABLED(CONFIG_OF_ADDRESS))
+		return;
+
 	np = of_find_node_by_path("/testcase-data/address-tests");
 	if (!np) {
 		pr_err("missing testcase data\n");
-- 
2.31.1

